src/index.ts

```diff
@@ -791,6 +791,21 @@ function ardrAddressEncoder(data: Buffer): string {
 
 }
 
+function scAddressEncoder(data:Buffer): string {
+  const hexString = data.toString('hex');
+  if(hexString.length !== 76) {
```

Comment:
A siacoin address is a hash + checksum, in hex representation. So aside from the length check, we should also add a checksum verification here. If you take a look at [this function](https://gitlab.com/NebulousLabs/Sia/-/blob/master/types/encoding.go#L910) it should become clear on what's going on. Both that function and the `String` function are of interest here, that's basically going from and to a siacoin address and string. Note that we use a blake2b hashing function.

Reply:
Thanks.